includes/admin/class-wc-admin-setup-wizard.php

```diff
@@ -132,8 +132,14 @@ public function setup_wizard() {
 			unset( $default_steps['shipping'] );
 		}
 
-		// Hide the activate step if Jetpack is already active.
-		if ( class_exists( 'Jetpack' ) && Jetpack::is_active() ) {
+		// Hide the activate step if Jetpack is already active, but not
+		// if we're returning from connecting Jetpack on WordPress.com.
+		if (
+			class_exists( 'Jetpack' ) &&
+			Jetpack::is_active() &&
+			! isset( $_GET['from'] ) &&
```

Comment:
You can replace it to:

```php
! isset( $_GET['from'], $_GET['activate_error'] )
```

Reply:
@claudiosanches required to merge or no?

Reply:
@jeffstieler nop, but will help a lot following our standards.

Reply:
Actually @claudiosanches - making the change you're requesting changes the logic. It needs to remain as-is.

Reply:
FWIW `! isset( $_GET['from'], $_GET['activate_error'] )` looks the same to me. 

> If multiple parameters are supplied then isset() will return TRUE only if all of the parameters are set

Reply:
@jeffstieler So based on that, if the logic above is not what you're after, can you double check the logic you have now?

Your code requires Jetpack to exist, be active, no `from`, and no `activate_error` right now.

Reply:
> So based on that, if the logic above is not what you're after, can you double check the logic you have now?
> 
> Your code requires Jetpack to exist, be active, no `from`, and no `activate_error` right now.

That logic is correct. What it's looking for is the case where Jetpack is already active when entering the wizard. The `from` param is set in the `wc_setup_activate_save()` method, which indicate that a redirect occurred from WPCOM back to the wizard. Having `activate_error` there is a big defensive, but nonetheless indicates that the user has previously been on the activate step.

Reply:
> FWIW `! isset( $_GET['from'], $_GET['activate_error'] )` looks the same to me.

To illustrate:

```php
php -a
Interactive shell

php > $test = array( 'a' => true, 'b' => null );
php > if ( ! isset( $test['a'] ) && ! isset( $test['b'] ) ) { echo 'not isset'; }
php > if ( ! isset( $test['a'], $test['b'] ) ) { echo 'not isset'; }
not isset
```